src/api/app/models/kiwi/image.rb

```diff
@@ -104,16 +104,17 @@ def to_xml
 
     return nil unless image && image.first_element_child
 
-    doc.xpath("image/packages").remove
-    xml_packages = package_groups.map(&:to_xml).join("\n")
-    image.first_element_child.after(xml_packages)
+    # for now we only write the default package group
+    xml_packages = default_package_group.to_xml
+    packages = doc.xpath('image/packages[@type="image"]').first
+    packages ? packages.replace(xml_packages) : image.last_element_child.after(xml_packages)
 
+    previous = doc.xpath("image/repository").first.try(:previous) || image.last_element_child
     doc.xpath("image/repository").remove
     xml_repos = repositories_for_xml.map(&:to_xml).join("\n")
-    image.first_element_child.after(xml_repos)
+    previous.after(xml_repos)
 
-    # Reparser for pretty printing
-    Nokogiri::XML(doc.to_xml, &:noblanks).to_xml
+    Nokogiri::XML(doc.to_xml, &:noblanks).to_xml(indent: kiwi_indentation(doc))
```

Comment:
Just wondering, would it be possible to use a .xml.builder view to generate this xml? Instead of Nokogiri and xpath? I'm finding it quite hard to read this code..

Reply:
Adding the indent parameter to doc.to_xml wouldn't have the same effect?

Reply:
@bgeuken would need to try that!

Reply:
@evanrolfe Which part do you mean? I doesn't seem that complicated.

Reply:
The whole method is hard to understand IMO, there are too many things going on. Its doing two things:
1. Deciding which things should go in the XML document.
2. Deciding how the XML document should be structured.

I think we should separate those two concerns just like how we separate models from views.

Reply:
You are right. I think we should move some code to separate methods, like setting kiwi_body.

Reply:
Moved to a dedicated Kiwi::Image::ToXml class!